src/protocol/modulus_conversion/convert_shares.rs

```diff
@@ -174,7 +182,14 @@ where
     try_join_all(
         zip(repeat(ctx), locally_converted_bits.iter())
             .enumerate()
-            .map(|(i, (ctx, row))| async move { convert_bit(ctx, RecordId::from(i), row).await }),
+            .map(|(i, (ctx, bit))| async move {
+                convert_bit(
+                    ctx.narrow(&ModulusConversion(i.try_into().unwrap())),
```

Comment:
Can you make the `ModulusConversion(` accept whatever this type is in order to avoid all of this casting?